statsd/src/main/java/com/avast/metrics/statsd/StatsDMetricsMonitor.java

```diff
@@ -256,4 +256,8 @@ protected String constructMetricName(Optional<String> finalName) {
             return a;
         }
     }
+
+    private String sanitizeNames(String name) {
+        return name.replaceAll("[^a-zA-Z0-9-]", "_");
```

Comment:
There are much more supported characters, do we want to be so strict? For example I remember I have already used `:` in product versions instead of `.` because they are visually similar, dot is separator.

Reply:
Hm, it looks like it is implementation specific. I have found that only alphanumerics, dots and underscores are supported.

Reply:
Sorry, I'm looking to the graphs, there are `-` at the moment, `:` was probably during development. There are a lot of statsd implementations in the wild, so maybe the strict variant is the best.

Reply:
I was searching a spec a year ago, but found nothing reliable.